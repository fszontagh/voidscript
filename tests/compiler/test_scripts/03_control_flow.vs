// Test control flow: if statements and loops
int $x = 10;

// If statement
if ($x > 5) {
    printnl("x is greater than 5");
} else {
    printnl("x is not greater than 5");
}

// While loop
int $i = 0;
printnl("While loop:");
while ($i < 3) {
    printnl("  i = ", $i);
    $i++;
}

// For loop
printnl("For loop:");
for (int $j = 0; $j < 3; $j++) {
    printnl("  j = ", $j);
}

// Nested control flow
printnl("Nested control flow:");
for (int $k = 0; $k < 3; $k++) {
    if ($k % 2 == 0) {
        printnl("  ", $k, " is even");
    } else {
        printnl("  ", $k, " is odd");
    }
}