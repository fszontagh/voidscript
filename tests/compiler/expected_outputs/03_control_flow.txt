x is greater than 5
While loop:
  i = 0
  i = 1
  i = 2
For loop:
  j = 0
  j = 1
  j = 2
Nested control flow:
  0 is even
  1 is odd
  2 is even